File: runinator-command-center/src/core/domain/models/artifact/workflow-run-artifact.ts
import type { JsonRecord } from "../../json";

export interface WorkflowRunArtifact {
  id: string;
  workflow_run_id: string;
  node_id: string;
  /** VM effect output has no legacy run_artifact id and therefore cannot use the old download route. */
  artifact_id: string | null;
  name: string;
  mime_type: string;
  size_bytes: number;
  uri: string;
  metadata?: JsonRecord;
  created_at: string;
}

File: runinator-models/src/workflow_runs/workflow_run_artifact.rs
#[allow(unused_imports)]
use super::*;

/// A run-level artifact declared by an output node, making it visible at workflow-run scope.
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct WorkflowRunArtifact {
    pub id: Uuid,
    pub workflow_run_id: Uuid,
    pub node_id: String,
    pub artifact_id: Uuid,
    pub name: String,
    pub mime_type: String,
    pub size_bytes: i64,
    pub uri: String,
    pub metadata: Value,
    pub created_at: DateTime<Utc>,
}
